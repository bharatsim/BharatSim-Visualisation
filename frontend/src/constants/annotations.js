export const areaAnnotationDirection = {
  HORIZONTAL: 'horizontal',
  VERTICAL: 'vertical',
};

export const rangeTypes = {
  NUMERIC: 'numeric',
  DATE: 'date',
};

export const DATE_FORMAT = 'yyyy-MM-dd';
