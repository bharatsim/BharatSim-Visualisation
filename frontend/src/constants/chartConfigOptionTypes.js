export const chartConfigOptionTypes = {
  X_AXIS: 'xAxis',
  Y_AXIS: 'yAxis',
};
