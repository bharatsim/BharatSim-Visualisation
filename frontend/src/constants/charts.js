export const chartTypes = {
  LINE_CHART: 'lineChart',
  BAR_CHART: 'barChart',
  PAI_CHART: 'paiChart',
  HEAT_MAP: 'heatMap',
  CHOROPLETH_MAP: 'choroplethMap',
  HISTOGRAM: 'histogram',
};

export const chartGroups = {
  'Line Chart': [chartTypes.LINE_CHART],
  'Bar Chart': [chartTypes.BAR_CHART],
  Histogram: [chartTypes.HISTOGRAM],
  'Geo Chart': [chartTypes.HEAT_MAP, chartTypes.CHOROPLETH_MAP],
};
