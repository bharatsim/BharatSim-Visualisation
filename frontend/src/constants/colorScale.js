export const scale = {
  0.2: '#88C1F7',
  0.4: '#1E3FD8',
  0.6: '#21C694',
  0.8: '#F7C75C',
  1: '#F37943',
};

// export const scale = {
//   0.2: 'blue',
//   0.4: 'cyan',
//   0.6: 'lime',
//   0.8: 'yellow',
//   1: 'red',
// };
