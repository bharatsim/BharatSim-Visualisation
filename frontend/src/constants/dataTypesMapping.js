const dataTypesMapping = {
  number: 'Number',
  string: 'String',
  date: 'Date',
  boolean: 'Boolean',
};
export default dataTypesMapping;
