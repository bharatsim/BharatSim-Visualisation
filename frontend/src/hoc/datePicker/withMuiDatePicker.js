import React from 'react';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import DateFnsUtils from '@date-io/date-fns';

export default (WrappedComponent) => (props) =>
  (
    <MuiPickersUtilsProvider utils={DateFnsUtils}>
      <WrappedComponent {...props} />
    </MuiPickersUtilsProvider>
  );
