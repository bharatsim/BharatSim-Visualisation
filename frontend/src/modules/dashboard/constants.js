export const AUTOSAVE_DEBOUNCE_TIME = 2000;
export const AUTOSAVE_ERROR_MESSAGE = 'Unable to save the dashboard';
