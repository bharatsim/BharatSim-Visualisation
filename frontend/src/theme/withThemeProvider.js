import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
