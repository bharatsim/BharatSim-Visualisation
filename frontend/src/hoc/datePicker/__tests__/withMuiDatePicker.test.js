import React from 'react';

import { DatePicker } from '@material-ui/pickers';
import { format } from 'date-fns';
import { render } from '@testing-library/react';

import withMuiDatePicker from '../withMuiDatePicker';
import withThemeProvider from '../../../theme/withThemeProvider';
import { DATE_FORMAT } from '../../../constants/annotations';

function DummyComponent() {
  return (
    <DatePicker
      label="Basic example"
      value={format(new Date(), DATE_FORMAT)}
      onChange={jest.fn()}
    />
  );
}

describe('withSnackBar', () => {
  const Component = withThemeProvider(withMuiDatePicker(DummyComponent));
  it('should provide date fns utils context to child component', () => {
    const { container } = render(<Component />);

    expect(container).toMatchSnapshot();
  });
});
