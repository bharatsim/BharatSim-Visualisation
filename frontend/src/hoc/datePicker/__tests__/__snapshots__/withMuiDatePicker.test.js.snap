// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withMuiDatePicker should provide date fns utils context to child component 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root"
    format="MMMM do"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
    >
      Basic example
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        readonly=""
        type="text"
        value="May 4th"
      />
    </div>
    
  </div>
</div>
`;
